src/esm_runscripts/workflow.py

```diff
@@ -1,8 +1,588 @@
+        """
+        Return the number of phases in workflow.
+        """
+        return len(self.phases)
+
+    def get_workflow_phase_by_name(self, phase_name):
+        """
+        Returns phase of phase_name
+
+        Arguments
+        ---------
+            self : class Workflow
+            phase_name : str (name of the phase to be returned
+
+        Returns
+        -------
+            phase : class phase or user_phase
+        """
+        for phase in self.phases + self.user_phases:
+            if phase.name == phase_name:
+                return phase
+
+    def get_phases_attribs_list(self, phase_type, attrib):
+        """
+        Returns a certain attribute for all phases as a list.
+
+        Parameters
+        ----------
+            self: class Workflow
+            phase_type: str (default or user)
+            attrib: str
+
+        Returns
+        -------
+            phases_attribs : list
+        """
+        if phase_type == 'user':
+            phases_attribs = [getattr(phase, attrib) for phase in self.user_phases]
+        else:
+            phases_attribs = [getattr(phase, attrib) for phase in self.phases]
+
+        return phases_attribs
+
+    def init_default_workflow(self, config):
+        """
+        Add workflow for precompute, compute, and tidy phases
+        etc information already here!
+
+        Parameters
+        ----------
+            self : Workflow object
+            config : dict
+
+        Returns
+        -------
+            self : Workflow object
+        """
+
+        workflow_phases = self.phases
+
+        # Calculating the number of tasks for each component/model
+        # needed for phase compute
+        tasks = calc_number_of_tasks(config)
+        # Initiate/create default workflow phase objects
+        # and reset/append to Workflow.phases variable
+        self.phases = []
+        for ind, phase in enumerate(workflow_phases):
+            self.phases.append(WorkflowPhase(phase))
+
+        for ind, phase in enumerate(self.phases):
+            if ind < self.num_phases - 1:
+                # Set run_before attrib of all phases (except last on)
+                # to the next phase name
+                phase.run_before = self.phases[ind+1].name
+                phase.next_submit.append(self.phases[ind+1].name)
+                phase.run_after = self.phases[ind-1].name
+            else:
+                # Set run_after attrib of last phase to previous phase name
+                phase.run_before = self.phases[0].name
+                phase.next_submit.append(self.phases[0].name)
+                phase.run_after = self.phases[ind-1].name
+
+            phase.cluster = phase.name
+            if phase.name in self.phases_to_submit_to_batch_system:
+                phase.batch_or_shell = 'batch'
+                phase.submit_to_batch_system = True
+                phase.run_on_queue = config["computer"]["partitions"]["compute"]["name"]
+                phase.nproc = tasks
```

Comment:
I suggest to delete this thing here and added as a dictionary in `configs/esm_software/esm_runscripts/defaults.yaml`.

IMO there is no reason to define the default workflow differently than the "user" workflow, it's better do define the default workflow somewhere else. There might be no need to make the individual phases be an object other than a dictionary. If you decide to make them an object (because for example you need to check certain things when you initialize the phase) then I'd say make it a subclass of dictionary.